components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionSecretProcessor.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.action.management;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.action.management.internal.ActionMgtServiceComponentHolder;
+import org.wso2.carbon.identity.action.management.model.AuthProperty;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementException;
+import org.wso2.carbon.identity.secret.mgt.core.model.ResolvedSecret;
+import org.wso2.carbon.identity.secret.mgt.core.model.Secret;
+import org.wso2.carbon.identity.secret.mgt.core.model.SecretType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.identity.action.management.constant.ActionMgtConstants.IDN_SECRET_TYPE_ACTION_SECRETS;
+
+/**
+ * Action secrets processor service implementation.
+ */
+public class ActionSecretProcessor {
+
+    private SecretType secretType;
+
+    public ActionSecretProcessor() {
+    }
+
+    public List<AuthProperty> encryptAssociatedSecrets(List<AuthProperty> authProperties, String actionId,
+                                                       String authType) throws SecretManagementException {
+
+        List<AuthProperty> encryptedAuthProperties = new ArrayList<>();
+        for (AuthProperty prop : authProperties) {
+            if (!prop.getIsConfidential()) {
+                encryptedAuthProperties.add(prop);
+                continue;
+            }
+            String secretName = buildSecretName(actionId, authType, prop.getName());
+            if (ActionMgtServiceComponentHolder.getInstance().getSecretManager()
+                    .isSecretExist(IDN_SECRET_TYPE_ACTION_SECRETS, secretName)) {
+                // Update existing secret property.
+                updateExistingSecretProperty(secretName, prop);
+            } else {
+                // Add secret to the DB.
+                if (StringUtils.isEmpty(prop.getValue())) {
+                    continue;
+                }
```

Comment:
IMO, this validation should not be done here.
And this applies for secret update as well.

Before coming here at the service layer the property value needs to be validated

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5795/commits/4c40292ff2e65531731f8e68897c74215dae5279